File: be/src/main/java/com/example/petclinic/dto/request/kennel/BookKennelRequest.java
package com.example.petclinic.dto.request.kennel;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class BookKennelRequest {
    String inTime;
    String outTime;
    String note;
    String doctorId;
    String petId;
    int kennelId;
}

File: ung_dung_thu_y/lib/dto/kennel/book_kennel_request.dart
class BookKennelRequest {
  final String inTime;
  final String outTime;
  final String? note;
  final String doctorId;
  final String petId;
  int kennelId;

  BookKennelRequest(
    this.inTime,
    this.outTime,
    this.note,
    this.doctorId,
    this.petId,
    this.kennelId,
  );

  Map<String, dynamic> toJson() => {
    'inTime': inTime,
    'outTime': outTime,
    'note': note ?? "",
    'doctorId': doctorId,
    'petId': petId,
    'kennelId': kennelId,
  };
}
